packages/tools/src/tools/CursorCrosshairSync.ts

```diff
@@ -0,0 +1,581 @@
+  _mouseOverElement: null | HTMLDivElement = null;
+  _currentMouseWorldPosition: null | Types.Point3 = null;
+  _currentCanvasPosition: null | Types.Point2 = null;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        shadow: true,
+        preventHandleOutsideImage: false,
+        displayThreshold: 5,
+        positionSync: true,
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  /**
+   * Overwritten mouseMoveCallback since we want to keep track of the current mouse position and redraw on mouseMove
+   * @virtual Event handler for Cornerstone MOUSE_MOVE event.
+   *
+   *
+   * @param evt - The normalized mouse event
+   * @param filteredAnnotations - The annotations to check for hover interactions
+   * @returns True if the annotation needs to be re-drawn by the annotationRenderingEngine.
+   */
+  mouseMoveCallback = (evt: EventTypes.MouseMoveEventType): boolean => {
+    const { detail } = evt;
+    const { element, currentPoints } = detail;
+    this.setMouseOverElement(element);
+    this._currentMouseWorldPosition = currentPoints.world;
+    //also need canvas postion for recalculating world position on stack change
+    this._currentCanvasPosition = currentPoints.canvas;
+
+    const annotation = this.getActiveAnnotation(element);
+    if (annotation === null) {
+      this.addNewAnnotation(evt);
+      return false;
+    }
+    this.updateAnnotationPosition(element, annotation);
+    return false;
+  };
+
+  //image change event seems to fire before image is rendered, so in order to get correct world position we need to wait for next render event
+  handleImageChange = (
+    evt:
+      | Types.EventTypes.StackViewportScrollEvent
+      | Types.EventTypes.VolumeNewImageEvent
+  ): void => {
+    const element = evt.target as HTMLDivElement;
+    element.addEventListener(
+      Enums.Events.IMAGE_RENDERED,
+      () => {
+        if (!this) return;
+        const viewport = getEnabledElement(element)?.viewport;
+        if (!viewport) return;
+        const renderingEngine = viewport.getRenderingEngine();
+        if (!renderingEngine) return;
+
+        //calculate new world position from chached canvas position
+        const activeAnnotation = this.getActiveAnnotation(element);
+
+        if (!this._currentCanvasPosition || !activeAnnotation) return;
+        const worldPos = viewport.canvasToWorld(this._currentCanvasPosition);
+        this._currentMouseWorldPosition = worldPos;
+
+        this.updateAnnotationPosition(element, activeAnnotation);
+      },
+      {
+        once: true,
+      }
+    );
+  };
+
+  //when the mouse is moved over a div, attach an event listener to this div to update the world position of the annotation when stack is scrolled
+  setMouseOverElement(element: HTMLDivElement | null): void {
+    if (element === this._mouseOverElement) return;
```

Comment:
also this, can we rename it from `mouseOverElement` to `elementWithCursor` or similar?